operator-framework-core/src/test/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverriderTest.java

```diff
@@ -25,13 +25,17 @@ void replaceNamedDependentResourceConfigShouldWork() {
     var dependents = configuration.getDependentResources();
```

Comment:
This test needs to be breaked down to multiple separate test cases. It's very hard to read this way.